Add import-symbol-at-point to package-fu

high-ego opened the pull request:
I wrote a function analogous to `export-symbol-at-point`. Unfortunately, while writing it, I rewrote a large part of the contrib. I would write tests to check if I broke something, but I'm not sure how to test this.

I did a little manual checking, and the only thing I think I broke is unexporting when the symbols are written as strings. The problem is that `slime-sexp-at-point` returns `nil` when it's called from a position like `("foo"|)`, with the point after a string that's the last element in a list. I don't know if this is expected behaviour, and I should write a wrapper that does what I need (return `"foo"`), or if I should change the function.

The messages that the new version produces are a little less informative. They are the same regardless of whether the defpackage form was changed. I'll change this if anyone thinks the distinction is important.

joaotavora commented:
high-ego <notifications@github.com> writes:

> I wrote a function analogous to export-symbol-at-point. Unfortunately, while writing it, I rewrote a large part of the contrib. I would write tests to check if I broke something, but I'm
> not sure how to test this.
>
> I did a little manual checking, and the only thing I think I broke is unexporting when the symbols are written as strings. The problem is that slime-sexp-at-point returns nil when it's
> called from a position like ("foo"|), with the point after a string that's the last element in a list. I don't know if this is expected behaviour, and I should write a wrapper that does
> what I need (return "foo"), or if I should change the function.
>
> The messages that the new version produces are a little less informative. They are the same regardless of whether the defpackage form was changed. I'll change this if anyone thinks the
> distinction is important.


FWIW, someone added this to SLY some years ago:

   https://github.com/joaotavora/sly/pull/147

and then I cleaned it up.  As far as I remember, didn't take a (big)
rewrite and nothing was broken.

Also, the author is fond of ASDF's "package-inferred system" and wrote a
sly-import-symbol-package-transform-function to deal with that, which I
don't use.

luismbo commented:
@high-ego can you have a look at João's commit and try to port things over, maybe, or double down on the need to rewrite.

high-ego commented:
I decided to stick with the rewrite, and wrote some tests for the parts I rewrote. To fix the case broken before, I modified `slime-bounds-of-sexp-at-point`.
